Reemit.Gui/ViewModels/Controls/HexEditor/HexEditorNavigationViewModel.cs

```diff
@@ -0,0 +1,122 @@
+﻿using AvaloniaHex.Document;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using Reemit.Gui.ViewModels.Navigation;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Linq;
+
+namespace Reemit.Gui.ViewModels.Controls.HexEditor;
+
+public class HexEditorNavigationViewModel : ReactiveObject
+{
+    private readonly HexEditorViewModel _hexEditorViewModel;
+
+    [Reactive]
+    public BitRange? NavigationBitRange { get; set; }
+
+    [ObservableAsProperty]
+    public HexNavigationRangeViewModel? ResolvedNavigationRange { get; set; }
+
+    [Reactive]
+    public ObservableCollection<HexNavigationRangeViewModel> NavigationRanges { get; set; } = [];
+
+    public ReactiveCommand<Unit, Unit> NextCommand { get; }
+
+    public ReactiveCommand<Unit, Unit> PreviousCommand { get; }
+
+    public HexEditorNavigationViewModel(HexEditorViewModel hexEditorViewModel)
+    {
+        _hexEditorViewModel = hexEditorViewModel;
+
+        this.WhenAnyValue(x => x.NavigationBitRange, x => x.NavigationRanges)
+            .Where(x => x.Item1 != null)
+            .Select(x =>
+            {
+                var (range, knownRanges) = x;
+
+                return knownRanges
+                    .Where(x =>
```

Comment:
Parameter `x` hides outer parameter with the same name here. We should probably rename it to avoid confusion. Something as simple as `y` instead of `x` works for me.

Reply:
Done.